conda_pack/scripts/windows/deactivate.bat

```diff
@@ -1,7 +1,11 @@
 @REM @ symbols in this file indicate that output should not be printed.
 @REM   Setting it this way allows us to not touch the user's echo setting.
 @REM   For debugging, remove the @ on the section you need to study.
-@REM
+@setlocal enabledelayedexpansion

```

Comment:
The problem with moving `@setlocal` above the `etc\conda\deactivate.d` loop is that environment variables that the deactivate scripts set (or unset) will be local, and not persisted to the user environment.

Reply:
Yep, I figured this out the hard way!